client/src/Components/Home/heplers.js

```diff
@@ -0,0 +1,26 @@
+const filterSkills = (arr, skillQuery) => {
+  return arr.filter(
+    item =>
+      !skillQuery.filter(
+        query => !item.skill.filter(_query => _query.id === query.id).length
+      ).length
+  );
+};
+
+const filterOfferTypes = (arr, offerTypeQuery) => {
+  return arr.filter(
+    item =>
+      !offerTypeQuery.filter(
+        query =>
+          !item.offer_type.filter(_query => _query.id === query.id).length
+      ).length
+  );
+};
+
+const filteredOffers = (value1, value2) => {
+  value1.filter(item => {
+    return value2.filter(_item => item.id === _item.id);
```

Comment:
@Fatmasiam I think you could replace this `value2.filter` with `value2.find` which might be a little faster (since it stops searching the rest of the array once it finds that first match)